### PR TITLE
Rs fixes

### LINKED ISSUE #4167: Townhall GUI ally teleport list lost on server restart
### Minecolonies version

Version: minecolonies-1.12.2-0.10.601-ALPHA

### Expected behavior

townhall GUI allways listing allies colonies to be able to teleport to them

### Actual behaviour

On a server start, the allied colonies list shows empty until yo visit them again by some way. no matter if the colony is active with the owner online

### Steps to reproduce the problem

1. start a dedicated server with at least 2 colonies, different owner
2. complete allied teleportation requisites and verify you can teleleport between colonies using townhall GUI at least one time
3. restart server...  no allied colonies listed
4. try with the 2 owners connected in theirs respective colonies... same problem

### PR DESCRIPTION
**Pull request by Raycoms**
Closes #4167

# Changes proposed in this pull request:
- Fix the quantity to be crafted
- Fixes the way we display allies/feuds
- Fix a bunch of lang issues
- Fix ladder pathing
- Fix problem when switching between singleplayer worlds where colony view data got messed up

Review please

### CODE REVIEW

**Review by someaddons — DISMISSED** [review]

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/api/java/com/minecolonies/api/colony/CompactColony.java  1
- src/main/java/com/minecolonies/coremod/colony/ColonyView.java  8
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4765031&bid=15884985)

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/api/java/com/minecolonies/api/colony/CompactColony.java  1
- src/main/java/com/minecolonies/coremod/colony/ColonyView.java  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4765031&bid=15884985)

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/api/java/com/minecolonies/api/colony/CompactColony.java  1
- src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/MovementHandler.java  15
- src/main/java/com/minecolonies/coremod/colony/ColonyView.java  10
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4765031&bid=15884985)

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/MovementHandler.java  15
- src/main/java/com/minecolonies/coremod/colony/ColonyView.java  10
- src/api/java/com/minecolonies/api/colony/CompactColony.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4765031&bid=15884985)

File: src/api/java/com/minecolonies/api/colony/CompactColony.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.minecolonies.api.colony;
+
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * Compact colony for allies and feud data.
+ */
+public class CompactColony
```
Comment (marchermans) [inline src/api/java/com/minecolonies/api/colony/CompactColony.java]: ColonyReference? Would probably be better.......

**Review by marchermans — DISMISSED** [review]

**Comment by reggaemuffin** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/colony/ColonyView.java  10
- src/api/java/com/minecolonies/api/colony/CompactColonyReference.java  1
- src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/MovementHandler.java  15
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4765031&bid=15884985)

**Comment by marvin-bitterlich** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/main/java/com/minecolonies/coremod/colony/ColonyView.java  10
- src/api/java/com/minecolonies/api/colony/CompactColonyReference.java  1
- src/main/java/com/minecolonies/coremod/entity/citizen/citizenhandlers/MovementHandler.java  15
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Minecolonies/minecolonies/pullRequest?prid=4765031&bid=15884985)

**Pull request merged**